modules/domain/lib/services/MapService.dart

```
@@ -1,23 +1,61 @@
-import 'package:domain/bloc/get_products/get_products_cubit.dart';
 import 'package:domain/repositories/targets_repository.dart';
+import 'package:geolocator/geolocator.dart';
+
+import '../bloc/map/map_cubit.dart';
 
 class MapService {
   final TargetsRepository _targetsRepository;
   final MapCubit _mapCubit;
+  double latitude = 0;
+  double longitude = 0;
 
   MapService(this._targetsRepository, this._mapCubit) {
     fetchTargets();
   }
 
   MapCubit get mapCubit => _mapCubit;
 
-  // TODO: implement fetchTargets
+  void addTarget(String title, double latitude, double longitude, double radius,
+      int topicId) async {
+    _mapCubit.isLoading();
+    final result = await _mapCubit.toCancelable(
+      _targetsRepository.addTarget(title, latitude, longitude, radius, topicId),
+    );
+    _mapCubit.onAddTargetResult(result);
+  }
+
   void fetchTargets() async {
     _mapCubit.isLoading();
     final result = await _mapCubit.toCancelable(
       _targetsRepository.getTargets(),
     );
-    _mapCubit.isSuccess([]);
-    //_mapCubit.onResult(result);
+    _mapCubit.onFetchTargetsResult(result);
+  }
+
+  Future<Position?> getLocation() async {
+    LocationPermission permission;
+    permission = await Geolocator.checkPermission();
+    if (permission == LocationPermission.denied) {
+      permission = await Geolocator.requestPermission();
+      if (permission == LocationPermission.denied) {
+        return Future.error('Location permissions are denied');
+      }
+    }
+
+    if (permission == LocationPermission.deniedForever) {
+      // Permissions are denied forever, handle appropriately.
+      return Future.error(
+          'Location permissions are permanently denied, we cannot request permissions.');
```

I'd return an error Type instead of a string.
check this:
https://github.com/rootstrap/flutter-modules/blob/35ed2f9ea41d6aa102cd5bd9c123ffce7c09428f/modules/geolocation/lib/service/GeolocationService.dart

---

looks good, thanks